src/views/preview/presentation.jsx

```diff
@@ -128,12 +130,28 @@ const PreviewPresentation = ({
                 message={embedCensorMessage(visibilityInfo.censorMessage)}
             />);
         }
-    } else if (canShare && !isShared) {
-        banner = (<Banner
-            actionMessage={<FormattedMessage id="project.share.shareButton" />}
-            message={<FormattedMessage id="project.share.notShared" />}
-            onAction={onShare}
-        />);
+    } else if (canShare) {
+        if (isShared) {
+            if (justShared) {
```

Comment:
Can't you just add justShared to the isShared if statement (i.e. `if (isShared && justShared) { ...`) to reduce the nesting and allow you to put the helpful comment that is currently at the bottom of this block alongside that condition?

Reply:
No, because the `else` for the unshared state (below) depends only on `isShared` and not on `justShared`. I did go back and forth on whether the way I have it now is better or worse than doing 

```
if (canShare && isShared && justShared) {
    if (isNewScratcher) {
        // ...
    } else {
        // ...
    }
} else if (canShare && !isShared) {
   // ...
}
```

...which is slightly more redundant, but maybe easier to read?